### DISCUSSION TITLE
Escaping the api/ requirement

### DISCUSSION THREAD

**Discussion by jayhale** [vercel/vercel#6187] [Help]
Is there a way to avoid the requirement that all functions exist in `api/`?

I want to deploy a very simple python project that has just `index.py` and `vercel.json`, i.e., no `api/` folder.  I get this error though, regardless of runtime selected in `vercel.json`:

```
Error: The pattern "index.py" defined in `functions` doesn't match any Serverless Functions inside the `api` directory. Learn More: https://vercel.link/unmatched-function-pattern
--
```

Is there a way to avoid this convention?

If not, what is the reason for enforcing this?

Thanks!

**Comment by fu-sen** [comment]
`index.py` file must always be in `api/`, but you can get rid of this with `rewrites`:
https://vercel.com/docs/configuration#project/rewrites

For example, `vercel.json`:
```
{
  "rewrites": [
    { "source": "/(.*)", "destination": "/api/index.py" }
  ]
}
```

**Reply by jayhale** [reply]
Thanks - what's the logic for enforcing the source tree structure?  The `api/` directory collides on a handful of projects I'd like to move to Vercel - any way to avoid having to use that specific directory?
